src/libs/Navigation/Navigation.js

```diff
@@ -259,6 +259,51 @@ function setIsNavigationReady() {
     resolveNavigationIsReadyPromise();
 }
 
+function navContainsConcierge(state) {
+    if (!state || !state.routeNames || !_.isArray(state.routeNames)) {
+        return false;
+    }
+
+    return _.includes(state.routeNames, SCREENS.CONCIERGE);
+}
+
+/**
+ * Waits for the navigation state to contain protected routes (specifically 'Concierge').
+ * If the navigation is in a state, where protected routes are available, the promise will resolve immediately.
```

Comment:
Just to make sure I understand, is "protected route" a new concept we're introducing? I'm not sure I understand what it has to do with Concierge specifically.

Reply:
nope, it just the way routes hidden behind the auth process are called. Since Concierge is the one of the most important screen that is available to users after logging in, I'm using it to detect the change and readines of the navigation state. In simple words - when Concierge is present as one of the available routes in the nav state, this means that app is showing protected routes.
Main cause of the whole bug was, that the app was trying to navigate to a protected route immediately after logging in, but the navigation was not ready yet and it was throwing an error.

Reply:
Is there another way we can wait for navigation to finish? I'm not sure I like the idea of checking if Concierge is an available route before allowing navigation. There might be situations in the future where the Concierge route is not available (I.e anonymous accounts)